ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/TPraos.hs

```diff
@@ -24,11 +26,17 @@ type instance BlockProtocol (ShelleyBlock c) = TPraos c
 instance TPraosCrypto c => BlockSupportsProtocol (ShelleyBlock c) where
   validateView _cfg (ShelleyHeader hdr _) = hdr
 
-  selectView _ (ShelleyHeader hdr _) = ChainSelectView
-    { csvChainLength = SL.bheaderBlockNo . SL.bhbody $ hdr
-    , csvIssuer      = SL.bheaderVk . SL.bhbody $ hdr
-    , csvIssueNo     = SL.ocertN . SL.bheaderOCert . SL.bhbody $ hdr
-    }
+  selectView _ (ShelleyHeader hdr _) =
+    ChainSelectView
+      { csvChainLength = SL.bheaderBlockNo . SL.bhbody $ hdr
+      , csvLeaderVRF =
+          SL.fromNatural . certifiedNatural
+            . SL.bheaderL
+            . SL.bhbody
+            $ hdr
```

Comment:
This relies on this code which seems fishy
```
-- | Our VRFAlgorithm provides a 'Natural', so we must exhibit an extractor to
--   use the bits as some other type
class VRFValue a where
  -- | Extract a value from a natural number derived from the VRF computation.
  fromNatural :: Natural -> a

instance VRFValue UnitInterval where
  -- TODO Consider whether this is a reasonable thing to do
  fromNatural k = truncateUnitInterval $ toInteger k % 10000
```
And truncate just does the remainder so would give weird results here.

TBH, I don't get why this thing needs a class.

Reply:
So it turns out this is ok, but the code is unclear. We've discussed improvements to make it clearer.

cc @JaredCorduan @nc6